rockcraft/services/image.py

```diff
@@ -0,0 +1,101 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2023 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Rockcraft Image Service."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from pathlib import Path
+from typing import cast
+
+from craft_application import AppMetadata, BaseService, ServiceFactory
+from craft_cli import emit
+
+from rockcraft import models, oci
+
+
+@dataclass
```

Comment:
Can we set this as frozen?

Reply:
done! I tried to set `RockcraftBuildInfo` also as frozen but then the base `BuildInfo` needs to be frozen too, which it probably right